pkg/apis/extensions/types.go

```diff
@@ -296,6 +300,17 @@ type DeploymentStatus struct {
 
 	// Total number of non-terminated pods targeted by this deployment that have the desired template spec.
 	UpdatedReplicas int `json:"updatedReplicas,omitempty"`
+
+	// Stuck indicates if the current deployment has failed.
+	Stuck bool `json:"stuck,omitempty"`
```

Comment:
After digging in other uses of `ConditionStatus` in the api I see that everywhere it is wrapped in a higher level `<Resource>Condition` so I guess we need to follow that pattern and have a `DeploymentCondition` here.